Podcast/PlayerViewController.swift

```diff
@@ -1,192 +1,134 @@
+        miniPlayerView.frame.size.width = view.frame.width
+        miniPlayerView.delegate = self
+        view.addSubview(miniPlayerView)
         
-        playPauseButton = UIButton(frame: .zero)
-        playPauseButton.setBackgroundImage(UIImage(named: "play_icon"), for: .normal)
-        playPauseButton.adjustsImageWhenHighlighted = false
-        playPauseButton.addTarget(self, action: #selector(playPauseButtonPress), for: .touchUpInside)
-        playPauseButton.frame = CGRect(x: 0, y: 0, width: PlayPauseButtonSize, height: PlayPauseButtonSize)
-        playPauseButton.center = CGPoint(x: view.frame.width/2, y: ControlButtonsCenterY)
-        controlsView.addSubview(playPauseButton)
+        episodeDetailView = EpisodeDetailView(frame: .zero)
+        episodeDetailView.frame.size.width = view.frame.width
+        episodeDetailView.frame.origin.y = playerHeaderView.frame.maxY
+        view.addSubview(episodeDetailView)
         
-        forwardsButton = UIButton(frame: .zero)
-        forwardsButton.setBackgroundImage(UIImage(named: "skip_forward_icon"), for: .normal)
-        forwardsButton.adjustsImageWhenHighlighted = false
-        forwardsButton.addTarget(self, action: #selector(forwardButtonPress), for: .touchUpInside)
-        forwardsButton.frame = CGRect(x: 0, y: 0, width: ForwardsBackwardsButtonSize, height: ForwardsBackwardsButtonSize)
-        forwardsButton.center = CGPoint(x: playPauseButton.frame.maxX + ControlButtonsCenterHorizontalOffset, y: ControlButtonsCenterY)
-        controlsView.addSubview(forwardsButton)
-        
-        forwardsLabel = UILabel(frame: .zero)
-        forwardsLabel.text = "30"
-        forwardsLabel.sizeToFit()
-        forwardsLabel.center = forwardsButton.center
-        controlsView.addSubview(forwardsLabel)
-        
-        backwardsButton = UIButton(frame: .zero)
-        backwardsButton.setBackgroundImage(UIImage(named: "skip_backward_icon"), for: .normal)
-        backwardsButton.adjustsImageWhenHighlighted = false
-        backwardsButton.addTarget(self, action: #selector(backwardButtonPress), for: .touchUpInside)
-        backwardsButton.frame = CGRect(x: 0, y: 0, width: ForwardsBackwardsButtonSize, height: ForwardsBackwardsButtonSize)
-        backwardsButton.center = CGPoint(x: playPauseButton.frame.minX - ControlButtonsCenterHorizontalOffset, y: ControlButtonsCenterY)
-        controlsView.addSubview(backwardsButton)
+        controlsView = PlayerControlsView(frame: CGRect(x: 0, y: 0, width: view.frame.width, height: 0))
+        controlsView.frame.origin.y = view.frame.height - controlsView.frame.size.height
+        view.addSubview(controlsView)
         
-        backwardsLabel = UILabel(frame: .zero)
-        backwardsLabel.text = "15"
-        backwardsLabel.sizeToFit()
-        backwardsLabel.center = backwardsButton.center
-        controlsView.addSubview(backwardsLabel)
+        updateUI()
         
         Player.sharedInstance.delegate = self
```

Comment:
the vc should tell the player what to do, not vice versa. In the current setup, you have the model updating the controller, it should be reversed.